sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -552,6 +552,39 @@ def test_polynomial_features_csr_X(deg, include_bias, interaction_only, dtype):
     assert_array_almost_equal(Xt_csr.A, Xt_dense)
 
 
+@pytest.mark.parametrize("columns", [1, 2, 3, 1000])
+def test_polynomial_features_csr_wide(columns):
+    x = sparse.csr_matrix(([1], ([0], [columns])))
+
+    est = PolynomialFeatures(3)
+    # this causes a crash with values of `columns` which are too large
+    est.fit_transform(x)
+
+
+@pytest.mark.parametrize(
```

Comment:
The usual way to do this would be with multiple parametrize decorators, one for each parameter.
However, I'm not sure that we need to test quite so many (1000) combinations. How long do they take to run altogether?

Reply:
```
pytest test_polynomial.py -k test_num_combinations
====================================================================================== test session starts =======================================================================================
platform linux -- Python 3.9.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/frederick/Projects/scikit-learn, configfile: setup.cfg
plugins: cov-2.11.1
collected 203 items / 139 deselected / 64 selected                                                                                                                                               

test_polynomial.py ................................................................                                                                                                        [100%]

=============================================================================== 64 passed, 139 deselected in 0.37s ===============================================================================
```

Reply:
On my machine it takes .37s to run all 64 combinations.

Reply:
Sorry, misread those ranges. It's still a fairly large runtime for a minor test, but acceptable.